### DISCUSSION TITLE
Default Addons

### DISCUSSION THREAD

**Discussion by etenzy** [kubermatic/kubeone#1889] [Q&A]
I've noticed that certain addons are being installed even though I haven't explicitly activated them.
I mean, it's great that ccm-hetzner, csi-hetzner, cni-canal, machinecontroller and metrics-server are automatically installed when the Hetzner provider is used, but unfortunately I couldn't find anything about it in the documentation.

Is there something like a default addon matrix?

```
{
  "kubeone": {
    "major": "1",
    "minor": "4",
    "gitVersion": "1.4.0",
    "gitCommit": "24a432834629b860257e4f3975ec223bc85a0c2f",
    "gitTreeState": "",
    "buildDate": "2022-02-16T11:04:51Z",
    "goVersion": "go1.17.5",
    "compiler": "gc",
    "platform": "darwin/arm64"
  },
  "machine_controller": {
    "major": "1",
    "minor": "43",
    "gitVersion": "v1.43.0",
    "gitCommit": "",
    "gitTreeState": "",
    "buildDate": "",
    "goVersion": "",
    "compiler": "",
    "platform": "linux/amd64"
  }
}
```

**Comment by kron4eg** [comment]
> Is there something like a default addon matrix?

Do you mean the matrix that will display under which provider which addons will be user? No, I don't think we have such info. The thing is, some addons are installed conditionally but that doesn't depend in the provider. For example machine-controller addons is regulated by the API field rather then provider.

**Comment by almereyda** [comment]
All this plays out in

https://github.com/kubermatic/kubeone/blob/master/pkg/addons/ensure.go

if you would like to learn more.
